pilot/pkg/model/validation.go

```diff
@@ -1863,8 +1863,28 @@ func ValidateVirtualService(msg proto.Message) (errs error) {
 	if len(virtualService.Hosts) == 0 {
 		errs = appendErrors(errs, fmt.Errorf("virtual service must have at least one host"))
 	}
+
+	allHostsValid := true
 	for _, host := range virtualService.Hosts {
-		errs = appendErrors(errs, validateHost(host))
+		if err := validateHost(host); err != nil {
+			errs = appendErrors(errs, validateHost(host))
```

Comment:
TODO: drop invalid hosts, add some public structure that can be queried by prometheus or /ctrlz to show invalid rules and hosts.

Having an invalid host in the set should not break the entire site - only the bad host.

Reply:
I can drop the invalid host as part of this pr if need be